cgrok/tree.c

```diff
@@ -0,0 +1,275 @@
+#include "tree.h"
+#include <stdio.h>
+#include <stdint.h>
+#include <string.h>
+
+// A shim to use dictlib trees instead of TC
+struct TCTREE {
+  dict *dict;
+  dict_itor *iter;
+};
+
+// Toyko cabinet accepts any arbitrary-sized data as a tree key.
+// We support this by using the first 4 bytes of the key to store the length.
+// So we have to wrap the libdict comparators with implementations that skip the first four bytes
```

Comment:
Does `libdict` use only 4 byte integers? If not, how does `dict_int_cmp` know the size of the keys?